src/ol_infrastructure/components/services/vault.py

```diff
@@ -134,3 +137,81 @@ def __init__(
                 default_ttl=db_config.default_ttl
             )
         self.register_outputs({})
+
+class OLVaultAWSSecretsEngineConfig(BaseModel):
+    app_name: Text
+    access_key: Text
+    default_lease_ttl_seconds: int = SIX_MONTHS
+    max_lease_ttl_seconds: int = SIX_MONTHS
+    description: Text
+    secret_key: Text
+    path: Text
+    policy_documents: Dict
```

Comment:
What kind of `Dict`? Would be good to specify the valid subtypes for the dictionary. e.g. `Dict[Text, Any]`, `Dict[Text, Text]`, etc. Would also be good to add a comment or validation function to specify what the constraints are.